progress_spinner.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"io"
+)
+
+type ProgressSpinner struct {
+	state                string
+	loadingMessage       string
+	updateLoadingMessage bool
+	writer               io.Writer
+}
+
+func NewProgressSpinner(writer io.Writer) *ProgressSpinner {
+	ipb := &ProgressSpinner{"|", "", true,writer}
+	return ipb
+}
+func (ps *ProgressSpinner) Next(loadingMessage string) {
+	var nextState string
+	switch ps.state {
+	case "|":
+		nextState = "/"
+		break
+	case "/":
+		nextState = "-"
+		break
+	case "-":
+		nextState = "\\"
+		break
+	default:
+		nextState = "|"
+	}
+	ps.state = nextState
+	if loadingMessage != ps.loadingMessage {
+		ps.loadingMessage = loadingMessage
+		ps.updateLoadingMessage = true
+	} else {
+		ps.updateLoadingMessage = false
+	}
+	ps.write()
```

Comment:
Let's remove the updateLoadingMessage flag:  
Either:
1. Replace Line 44 with line 35, remove lines 35-40 and remove updateLoadingMessage member. Pass the loading message into write()  and update the message if there's a difference. ; or
2. Remove lines 35-41 and remove updateLoadingMessage member. Remove the write()[since it's not being used at the moment] and move its content where lines 35-41 use to be.  Line 44 would need to be updated to reflect what was previously at line 35.